hardy/src/main.rs

```diff
@@ -117,12 +122,11 @@ fn main() {
 
     let mut rng = thread_rng();
 
-    let population = Population::new(population_size,
+    let mut population = Population::new(population_size,
```

Comment:
The signatures of `mate()` and `swap_generations()` each required a `&mut Population`.

- `let population...` is immutably owned
- `let mut population` is mutably owned.